addons/website_sale_loyalty/models/loyalty_rule.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+class LoyaltyRule(models.Model):
+    _inherit = 'loyalty.rule'
+
+    website_id = fields.Many2one(related='program_id.website_id', store=True)
+
+    # NOTE: is this sufficient?
```

Comment:
Do you think about other constraints ?

Reply:
The issue here is that since it is not an sql constraint it can definitely be bypassed using archive etc.
IMO it is sufficient, as we will only fetch one anyway in case of duplicated key.